crates/extensions/tedge_mqtt_ext/src/lib.rs

```diff
@@ -345,6 +365,30 @@ impl FromPeers {
                             client.unsubscribe_many(diff.unsubscribe).await.unwrap();
                         }
                     });
+                    if !tf.is_empty() {
```

Comment:
I'm wondering if this would meet the dynamic subscription requirements of the dynamic mapper, where in most cases when new pipelines are dynamically added, they are most likely going to subscribe to already subscribed topics like `te/+/+/+/+/m/meta` and would be expecting the retained messages from those topics. But, since dynamic connections are established only when a new subscription is added or an existing one is upgraded, the newly added pipelines won't get the existing retained messages as far as I understand. This won't meet our requirement, right? I'm just wondering if blindly establishing a new connection would be better here.

Reply:
> No, the exact opposite. We establish a dynamic connection when we receive a subscription that does already exist.

Not sure what I was doing wrong then. When I tested the same by tweaking the `dynamic_subscribers_receive_retain_messages` test with the following code at the end:

```
client_0
        .send(PublishOrSubscribe::Subscribe(SubscriptionRequest {
            diff: SubscriptionDiff {
                subscribe: ["a/b".into()].into(),
                unsubscribe: [].into(),
            },
            client_id: client_id_1,
        }))
        .await
        .unwrap();
assert_eq!(timeout(client_0.recv()).await.unwrap(), msg);
```

the assertion validating if `client0` gets the same retained message again, timed out.

Reply:
That is the intention. A new client subscribing shouldn't redeliver the retain messages to an existing subscription?

Reply:
I think the mapper should be using both (if we change the input topic for a mapping rule, for instance, that will cause a change in the subscription, but shouldn't need a new MQTT actor connection).

Also, the "connection" part has nothing to do with MQTT connections and is merely a dynamic connection between two actors.